lib/class-wp-theme-json.php

```diff
@@ -0,0 +1,953 @@
+			),
+		),
+		'settings' => array(
+			'color'      => array(
+				'custom'         => null,
+				'customGradient' => null,
+				'gradients'      => null,
+				'link'           => null,
+				'palette'        => null,
+			),
+			'spacing'    => array(
+				'customPadding' => null,
+				'units'         => null,
+			),
+			'typography' => array(
+				'customFontSize'   => null,
+				'customLineHeight' => null,
+				'dropCap'          => null,
+				'fontFamilies'     => null,
+				'fontSizes'        => null,
+				'fontStyles'       => null,
+				'fontWeights'      => null,
+				'textDecorations'  => null,
+				'textTransforms'   => null,
+			),
+			'custom'     => null,
+		),
+	);
+
+	/**
+	 * Presets are a set of values that serve
+	 * to bootstrap some styles: colors, font sizes, etc.
+	 *
+	 * They are a unkeyed array of values such as:
+	 *
+	 * ```php
+	 * array(
+	 *   array(
+	 *     'slug'      => 'unique-name-within-the-set',
+	 *     'name'      => 'Name for the UI',
+	 *     <value_key> => 'value'
+	 *   ),
+	 * )
+	 * ```
+	 *
+	 * This contains the necessary metadata to process them:
+	 *
+	 * - path          => where to find the preset in a theme.json context
+	 *
+	 * - value_key     => the key that represents the value
+	 *
+	 * - css_var_infix => infix to use in generating the CSS Custom Property. Example:
+	 *                   --wp--preset--<preset_infix>--<slug>: <preset_value>
+	 *
+	 * - classes      => array containing a structure with the classes to
+	 *                   generate for the presets. Each class should have
+	 *                   the class suffix and the property name. Example:
+	 *
+	 *                   .has-<slug>-<class_suffix> {
+	 *                       <property_name>: <preset_value>
+	 *                   }
+	 */
+	const PRESETS_METADATA = array(
+		array(
+			'path'          => array( 'settings', 'color', 'palette' ),
+			'value_key'     => 'color',
+			'css_var_infix' => 'color',
+			'classes'       => array(
+				array(
+					'class_suffix'  => 'color',
+					'property_name' => 'color',
+				),
+				array(
+					'class_suffix'  => 'background-color',
+					'property_name' => 'background-color',
+				),
+			),
+		),
+		array(
+			'path'          => array( 'settings', 'color', 'gradients' ),
+			'value_key'     => 'gradient',
+			'css_var_infix' => 'gradient',
+			'classes'       => array(
+				array(
+					'class_suffix'  => 'gradient-background',
+					'property_name' => 'background',
+				),
+			),
+		),
+		array(
+			'path'          => array( 'settings', 'typography', 'fontSizes' ),
+			'value_key'     => 'size',
+			'css_var_infix' => 'font-size',
+			'classes'       => array(
+				array(
+					'class_suffix'  => 'font-size',
+					'property_name' => 'font-size',
+				),
+			),
+		),
+		array(
+			'path'          => array( 'settings', 'typography', 'fontFamilies' ),
+			'value_key'     => 'fontFamily',
+			'css_var_infix' => 'font-family',
+			'classes'       => array(
```

Comment:
All the presets after this line and including it. Should not include classes. Only colors, gradients, and font sizes have classes.

Reply:
Ah, I missed font families don't generate classes. However, all the others do, as per what's in master https://github.com/WordPress/gutenberg/blob/master/lib/global-styles.php#L687 Updated (and added test!) at https://github.com/WordPress/gutenberg/pull/26803/commits/c0a958d20ea549016ff8758385f5a4dd87fba5fc

Reply:
Well nice catch, the others should also not generate classes as we decided not to use classes anymore besides for things that already had them. But that should be fixed in a different PR.